src/components/AddressForm.tsx

```diff
@@ -1,109 +1,136 @@
-import {CONST as COMMON_CONST} from 'expensify-common/lib/CONST';
-import lodashGet from 'lodash/get';
-import PropTypes from 'prop-types';
+import type {CONST as COMMON_CONST} from 'expensify-common/lib/CONST';
 import React, {useCallback} from 'react';
 import {View} from 'react-native';
-import _ from 'underscore';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import Navigation from '@libs/Navigation/Navigation';
 import * as ValidationUtils from '@libs/ValidationUtils';
 import CONST from '@src/CONST';
+import type {OnyxFormKey} from '@src/ONYXKEYS';
+import type {AddressForm as AddressFormValues} from '@src/types/onyx';
 import AddressSearch from './AddressSearch';
 import CountrySelector from './CountrySelector';
 import FormProvider from './Form/FormProvider';
 import InputWrapper from './Form/InputWrapper';
 import StatePicker from './StatePicker';
 import TextInput from './TextInput';
 
-const propTypes = {
+type AddressFormProps = {
     /** Address city field */
-    city: PropTypes.string,
+    city?: string;
 
     /** Address country field */
-    country: PropTypes.string,
+    country?: keyof typeof CONST.COUNTRY_ZIP_REGEX_DATA | '';
 
     /** Address state field */
-    state: PropTypes.string,
+    state?: keyof typeof COMMON_CONST.STATES | '';
 
     /** Address street line 1 field */
-    street1: PropTypes.string,
+    street1?: string;
 
     /** Address street line 2 field */
-    street2: PropTypes.string,
+    street2?: string;
 
     /** Address zip code field */
-    zip: PropTypes.string,
+    zip?: string;
 
     /** Callback which is executed when the user changes address, city or state */
-    onAddressChanged: PropTypes.func,
+    onAddressChanged?: (data: string, key: string) => void;
 
     /** Callback which is executed when the user submits his address changes */
-    onSubmit: PropTypes.func.isRequired,
+    onSubmit: () => void;
 
     /** Whether or not should the form data should be saved as draft */
-    shouldSaveDraft: PropTypes.bool,
+    shouldSaveDraft?: boolean;
 
     /** Text displayed on the bottom submit button */
-    submitButtonText: PropTypes.string,
+    submitButtonText?: string;
 
     /** A unique Onyx key identifying the form */
-    formID: PropTypes.string.isRequired,
+    formID: OnyxFormKey;
 };
 
-const defaultProps = {
-    city: '',
-    country: '',
-    onAddressChanged: () => {},
-    shouldSaveDraft: false,
-    state: '',
-    street1: '',
-    street2: '',
-    submitButtonText: '',
-    zip: '',
+type ValidatorErrors = {
+    addressLine1?: string;
+    city?: string;
+    country?: string;
+    state?: string;
+    zipPostCode?: string | Array<string | Record<string, string>>;
```

Comment:
```suggestion
    zipPostCode?: Record<string, Localize.MaybePhraseKey>;
```

Could you test if this works?

Reply:
Updated